data/lua/core/_initial.lua

```diff
@@ -7,19 +7,24 @@ print("Loading Lua core files...")
 local _ = wesnoth.textdomain "wesnoth"
 
 -- Marks a function or subtable as deprecated.
--- Parameters:
----- elem_name: the full name of the element being deprecated (including the module)
----- replacement: the name of the element that will replace it (including the module)
----- level: deprecation level (1-4)
----- version: the version at which the element may be removed (level 2 or 3 only)
----- Set to nil if deprecation level is 1 or 4
----- elem: The actual element being deprecated, ignored if level is 4
----- detail_msg: An optional message to add to the deprecation message
-function wesnoth.deprecate_api(elem_name, replacement, level, version, elem, detail_msg)
+---@generic T
+---@param elem_name string the full name of the element being deprecated (including the module)
+---@param replacement_name string the name of the element that will replace it (including the module)
```

Comment:
Might be helpful to say which of these strings have a functional effect, and which are only used in text printed to the user / UMC author.

Some of the code handles replacement_name being empty or nil, but other parts seem to expect it to be non-empty.